rustls/src/lib.rs

```diff
@@ -487,6 +502,23 @@
 #![cfg_attr(bench, feature(test))]
 #![no_std]
 
+// This constraint is also enforced by `portable-atomic` crate - enforcing here as well
+// for extra clarity (with a QUICK WORKAROUND)
+// FOR FUTURE CONSIDERATION: it may be a nicer developer experience if both this crate and
+// `portable-atomic` would allow both of these features in some form, which would provide
+// some form a fallback in case `critical-section` feature is not specified - this is now
+// tracked in: https://github.com/brody4hire/portable-rustls/issues/27
```

Comment:
I am undecided about this idea at this point, maybe just remove this text & track my further thoughts in #27